dump_manager_bmc.cpp

```diff
@@ -40,33 +40,70 @@ void Manager::create(Type type, std::vector<std::string> fullPaths)
 sdbusplus::message::object_path
     Manager::createDump(phosphor::dump::DumpCreateParams params)
 {
-    if (!params.empty())
+    if (params.size() > 1)
     {
-        log<level::WARNING>("BMC dump accepts no additional parameters");
+        log<level::WARNING>(
+            "BMC dump accepts not more than 1 additional parameter");
     }
+
     std::vector<std::string> paths;
     auto id = captureDump(Type::UserRequested, paths);
 
     // Entry Object path.
     auto objPath = std::filesystem::path(baseEntryPath) / std::to_string(id);
 
     std::time_t timeStamp = std::time(nullptr);
-    createEntry(id, objPath, timeStamp, 0, std::string(),
+
+    // Get the generator id from params
+    using InvalidArgument =
```

Comment:
it is better to throw error before calling the create

Reply:
In line 71, we do log it and proceed with creating the entry in line 90.
Also if there is an error in retrieving, line 79 will take care of the same.

Am i missing something here?

Reply:
Done